File: test-suite-function-client-aws-kotlin/src/test/kotlin/io/micronaut/docs/function/aws/AnalyticsClientTest.kt
package io.micronaut.docs.function.aws

import io.micronaut.context.ApplicationContext
import io.micronaut.function.client.FunctionDefinition
import io.micronaut.function.client.aws.AWSInvokeRequestDefinition
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

@MicronautTest(startApplication = false)
internal class AnalyticsClientTest {
    @Inject
    lateinit var applicationContext: ApplicationContext
    @Test
    fun testSetupFunctionDefinitions() {
        val definitions = applicationContext.getBeansOfType(FunctionDefinition::class.java)
        Assertions.assertEquals(1, definitions.size)
        Assertions.assertTrue(definitions.stream().findFirst().isPresent)
        Assertions.assertTrue(definitions.stream().findFirst().get() is AWSInvokeRequestDefinition)
        val invokeRequestDefinition = definitions.stream().findFirst().get() as AWSInvokeRequestDefinition
        Assertions.assertEquals("analytics", invokeRequestDefinition.name)
        //Assertions.assertEquals("AwsLambdaFunctionName", invokeRequestDefinition.invokeRequest.functionName)
    }
}

File: test-suite-python/src/test/python/micronaut/docs/function/aws/AnalyticsClientTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import ApplicationContext
from micronaut.function.client import FunctionDefinition
from micronaut.function.client.aws.v2 import AwsInvokeRequestDefinition
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from micronaut.docs.function.client.aws.atnamed.AnalyticsClient import AnalyticsClient as AtNamedAnalyticsClient
from micronaut.docs.function.client.aws.methodnamed.AnalyticsClient import AnalyticsClient as MethodNamedAnalyticsClient


@MicronautTest(startApplication=False)
class AnalyticsClientTest:
    application_context: Annotated[ApplicationContext, Inject]

    @Test
    def test_setup_function_definitions(self):
        definitions = self.application_context.getBeansOfType(FunctionDefinition)

        assert len(definitions) == 1
        definition = definitions.iterator().next()
        assert isinstance(definition, AwsInvokeRequestDefinition)
        assert definition.getName() == "analytics"

    @Test
    def test_function_clients_are_beans(self):
        assert self.application_context.containsBean(AtNamedAnalyticsClient)
        assert self.application_context.containsBean(MethodNamedAnalyticsClient)
